### DISCUSSION TITLE
Maybe even cooler?

### DISCUSSION THREAD

**Discussion by kisstheu** [sqlitebrowser/sqlitebrowser#3103] [General]
Will there be a QML version in the future? Maybe it would be a little cooler? : )

**Comment by scottfurry** [comment]
Understand that this is not the opinion of the DB4S project but my own personal observations.

QML is designed to have a visual "wow" factor. IMO, it is a perpetuation of the idea to push the web onto the desktop environment. Automotive and small screen developers jump on the QML to leverage simple display with touch screen capabilities. For a desktop environment with multi-purpose editing role, QML seems like overkill. Having looked at the documentation to develop in QML and playing with small example apps, I can only say *yuck*.

A back end would still be required in order to interface component libraries(SQLite, JSON, et al). The task of pushing the interface, and separating out non-interface logic, would be a task verging on Herculean. And that's before any other conversation about layouts, user inputs, or signals/slots connections. I have not even addressed the issue of what to do with the old code and the auto-building being done for DB4S releases.

My impression of the feature request, if we can call this as such, is that the visual "wow" is not worth the effort of porting. There would have to be an incredibly compelling reason to justify the vast expenditure of resources(think people banging on keyboards and testing code pathways) before considering going down this road. It would, in effect, discard large parts of existing code.

However, if you want to try to build one... I won't stop you. Understand if I, or others, knowingly do not share your enthusiasm.

**Reply by chrisjlocke** [reply]
![image](https://user-images.githubusercontent.com/7872218/180935157-0b0df433-0ad8-432a-9019-0574cac44f0d.png)

As the song goes, "Be our guest".  With >500 issues still to address, there is no way the current development team would even remotely possibly even lightly entertain this.  To what advantage?

**Reply by kisstheu** [reply]
Thank you very much for your reply. Indeed, as you said, the goal of QML is to compete with the embedded market, and the background logic cannot use snail like js.
Maybe many, many years later, qml+widgets=? A new framework was born (*^▽^*)
